### PR TITLE
# Added Aws4SignedHttpConnectionFactory to sign elasticsearch requests

### PR DESCRIPTION
**Pull request by CCristi**

### CODE REVIEW

File: src/deep-search/lib/Client/Connection/Aws4SignedHttpConnectionFactory.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            ...params,
+            service: 'es'
+        });
+
+        params = super.makeReqParams(params);
+        params.headers = params.headers || {};
+
+        ['X-Amz-Date', 'X-Amz-Security-Token', 'Authorization'].forEach((header) => {
+          params.headers[header] = aws4Signature.headers[header];
+        });
```
Comment (devs-deep) [inline src/deep-search/lib/Client/Connection/Aws4SignedHttpConnectionFactory.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/MitocGroup/deep-framework/file/2319638134/issues/source?bid=2895475&fileBranchId=3204331#l47)

Comment (devs-deep) [inline src/deep-search/lib/Client/Connection/Aws4SignedHttpConnectionFactory.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/MitocGroup/deep-framework/file/2320249722/issues/source?bid=2895475&fileBranchId=3204331#l47)

File: src/deep-log/lib/Driver/RumSqsDriver.js [hunk 2]
```diff
@@ -252,6 +252,23 @@ export class RumSqsDriver extends AbstractDriver {
...
+   * @param {Object[]} additionalAttributes
+   */
+  getQueueAttributes(callback, additionalAttributes = []) {
+    let params = {
+      QueueUrl: this.queueUrl,
+      AttributeNames: [
+        'ApproximateNumberOfMessages',
+        'ApproximateNumberOfMessagesNotVisible',
+        'ApproximateNumberOfMessagesDelayed'
+      ].concat(additionalAttributes)
```
Comment (mgoria) [inline src/deep-log/lib/Driver/RumSqsDriver.js]: - please use `merge` instead of `concat`

File: src/deep-log/lib/Driver/RumSqsDriver.js [hunk 3]
```diff
@@ -264,4 +281,18 @@ export class RumSqsDriver extends AbstractDriver {
 
     return regionParts[1];
   }
+
+  /**
+   * @returns {String}
+   */
+  static get ES_LOGS_INDEX() {
+    return 'sqs';
```
Comment (mgoria) [inline src/deep-log/lib/Driver/RumSqsDriver.js]: - please rename index to `rum`

File: src/deep-search/lib/Client/Connection/Aws4SignedHttpConnectionFactory.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+   * @param {Object} aws4Signature
+   * @returns {$ES6_ANONYMOUS_CLASS$}
+   */
+  create() {
+    let connectionFactory = this;
+
+    /**
+     * Aws4 Signed Http Connection
+     */
+    return class extends HttpConnection {
```
Comment (mgoria) [inline src/deep-search/lib/Client/Connection/Aws4SignedHttpConnectionFactory.js]: - Create a separate class , e.g `HttpConnection `

**Pull request closed without merging**